src/Resources/contao/dca/tl_page.php

```diff
@@ -28,7 +28,8 @@
 			array('tl_page', 'addBreadcrumb'),
 			array('tl_page', 'setRootType'),
 			array('tl_page', 'showFallbackWarning'),
-			array('tl_page', 'makeRedirectPageMandatory')
+			array('tl_page', 'makeRedirectPageMandatory'),
+			array('tl_page', 'addListDragAndDrop')
```

Comment:
Can you explain why this is needed here? Can't we find a general solution for `DC_Table` so that code is just added for any DCA with the tree view?

Reply:
I decided this way because the DCTable monster is used everywhere and the selectors differs a little bit. It was the fastest way to bring up the feature and discuss it. I prefer the better done than perfect philosophy :)

Perhaps we can add an option for the DCA`s to configure it?

Reply:
Yeah it's just a problem right now for my own tree view DCA's. They all would need to implement that callback. What's specific there? The `listingContainer`?

Reply:
Just the droppables. Because of the current markup everything is an tl_folder but on the articles we just wanna allow to drop into pages or after articles inside a pages. So that .level_0 selector comes up :/

Reply:
Yeah so maybe we can think about some general - new - droppable container with the correct class and just make sure the `DC_Table` adds that. Some data attribute or so, which will then work everywhere. That would be excellent :)

Reply:
Also the current url is generated here to trigger the location.href.

Reply:
@Toflar i like to implement it as option of the dca list like $GLOBALS['TL_DCA']['tl_page']['list']['dragdrop'] = [ ..options ] or underneath sorting. What did you think? What @leofeyer think?

Reply:
I don't understand what needs to be configurable yet. To me, whenever a configuration can be omitted, it should :)

Reply:
I understand your intention but the feature was just created as quick win to work for site structure and article overview. I like the idea to implement into any tree but we didn`t know if it works in any case. So i think it should be explicit activated via dca.

By the way, i thankfully realized with this discussion that i`ve missed to check the user rights ☠️. Only if the user has the right to move items their could use this feature. This maybe belongs to #1394 too. @ausi did you add an check for the file drag and drop?

Maybe we should talk about this topics via mumble in the upcoming days?

Reply:
Yes, we should. The next call is next week on Thursday, April 12th.